source/mainnet/smart-contracts/tutorials/smartContractUpgrade/smartContractUpgrade.rst

```diff
@@ -0,0 +1,287 @@
+
+.. code-block:: console
+
+    $concordium-client module deploy ./smart_contract_upgrade.wasm.v1 --sender <YourAccountAddress> --grpc-port 10000 --grpc-ip node.testnet.concordium.com
+
+.. image:: ./images/deploy_contract_version2.png
+   :alt: Deploy contract version 2
+   :align: center
+
+The ``moduleReference`` returned when deploying ``contract-version2`` is different from the ``moduleReference`` when deploying ``contract-version1``.
+This shows that the module's code is not identical.
+
+Create the ``upgrade.json`` file with the following content by inserting the ``moduleReference`` returned by the previous step:
+
+.. code-block:: json
+
+   {
+      "migrate": {
+         "Some": [
+            [
+            "migration",
+            ""
+            ]
+         ]
+      },
+      "module": "<ModuleReferenceContractVersion2>"
+   }
+
+For example, when using the module reference from the above screenshot, the file will look as follows:
+
+.. code-block:: json
+
+   {
+      "migrate": {
+         "Some": [
+            [
+            "migration",
+            ""
+            ]
+         ]
+      },
+      "module": "31539c983f2ee56822041230d7fd20a3516da9271837e23bb77111bb8c4c7dcd"
+   }
+
+When invoking the ``upgrade`` function with the above file, it specifies that the smart contract
+code should be upgraded to the module ``31539c983f2ee56822041230d7fd20a3516da9271837e23bb77111bb8c4c7dcd``
+and the ``migration`` function should be invoked
+with no input parameters (empty string: ``""``) as part of the upgrade.
+
+Upgrade the smart contract as follows:
+
+.. code-block:: console
+
+   $concordium-client contract update <YourContractVersion1Index> --entrypoint upgrade --parameter-json upgrade.json --energy 5000 --sender <YourAccountAddress> --grpc-port 10000 --grpc-ip node.testnet.concordium.com
+
+For example, when using the smart contract index from the above screenshot, the command will look as follows:
+
+.. code-block:: console
+
+   $concordium-client contract update 4462 --entrypoint upgrade --parameter-json upgrade.json --energy 5000 --sender <YourAccountAddress> --grpc-port 10000 --grpc-ip node.testnet.concordium.com
+
+You should see the following output.
+
+.. image:: ./images/upgrade.png
+   :alt: Upgrade contract version 1
+   :align: center
+
+You can check that the state migration was successful by inspecting the state with the view function:
+
+.. code-block:: console
+
+   $concordium-client contract invoke <YourContractVersion1Index> --entrypoint view --grpc-port 10000 --grpc-ip node.testnet.concordium.com
+
+.. image:: ./images/view_contract_version2.png
+   :alt: View contract version 2
+   :align: center
+
+If the migration function had a bug and changed the shape of the state wrongly, your state might be corrupted. You might see different output errors in case of a corrupted state. One example is shown below:
```

Comment:
```suggestion
If the migration function had a bug and changed the shape of the state incorrectly, your state might be corrupted. You might see different output errors in case of a corrupted state. One example is shown below:
```